### PR TITLE
Usability improvements

### PR DESCRIPTION
**Pull request by unflxw**
### [Use fake revision if none given](https://github.com/appsignal/frontend-test-setups/commit/cc18c578dcf709834659ec0f17d5dfef15c2375c)

When using this to test sourcemap behavior, you want to see a
different revision uploaded each time, but you don't actually care
what the revision name _is_, and having to set it manually to a
different value in each run is mildly inconvenient.

If no revision value is given, generate something that looks like
a Git commit SHA with a "-fake" prefix, and use that as a revision.

### [Add appsignal to the global object](https://github.com/appsignal/frontend-test-setups/commit/ea2e01fe6e31e2762197a88793942b1a11d36e1a)

This allows one to easily verify that it's configured correctly
and to manually trigger sending errors.

### [Improve README](https://github.com/appsignal/frontend-test-setups/pull/36/commits/fa0e9ac5012d53fae54294622e6ef09716f2c291)

Move automated tests dependencies (geckodriver) to the automated
tests section.

Clarify that `yarn` must be installed.

Clarify that the Node.js version must be Node 16, using a
`.tool-versions` file to set it automatically for `asdf` users.

### [Remove Angular analytics prompt](https://github.com/appsignal/frontend-test-setups/pull/36/commits/31e8271db4b221fe5336aba33a6b857197e7530a)

An annoying prompt about analytics blocks the test run until it
is answered. Disable it.

### CODE REVIEW

File: support/templates/appsignal.js.erb [hunk 1]
```diff
@@ -2,8 +2,12 @@
...
-export default new Appsignal({
+const appsignal = new Appsignal({
   key: "<%= @frontend_key %>",
   revision: "<%= @revision %>",
   uri: "<%= @uri %>"
 });
+
+export default appsignal;
+
+globalThis.appsignal = appsignal;
```
Comment (luismiramirez) [inline support/templates/appsignal.js.erb:13]: I'm not sure if this fits for `globalThis`, but what about calling the global property `__APPSIGNAL__` as in the Node.js integration?

**Review by luismiramirez — APPROVED** [review]

File: support/templates/appsignal.js.erb (hunk 1 above)
Comment (unflxw) [reply support/templates/appsignal.js.erb:13]: To be clear, this is only for our convenience, to inspect the object and play around with it manually, from the browser's console in the test setup. I don't mind calling it `__APPSIGNAL__`, but the idea is for it to be easy to remember, so you just open the console and go `appsignal.sendError()` and it works.

We _may_ want to make the client globally accessible in the actual integration tho 🤔

File: support/helpers.rb [hunk 2]
```diff
@@ -57,6 +57,10 @@ def get_keys
   YAML.load_file("keys.yml")
 end
 
+def fake_revision
+  "0123456789abcdef".chars.sample(7).join + "-fake"
```
[the remark this answers is not in the document]
Comment (unflxw) [reply support/helpers.rb:61]: Sounds good! The idea is to make it clear that this wasn't meant to be a real revision.

File: support/templates/appsignal.js.erb (hunk 1 above)
Comment (luismiramirez) [reply support/templates/appsignal.js.erb:13]: +1 to both things.

- Keeping it `.appsignal`
- Adding it to the actual integration.

I'm not sure it'll be as useful as in the Node.js integration but is good to have it for the sake of consistency.

**Review by shairyar — APPROVED** [review]

**Review by tombruijn — APPROVED** [review]

File: support/templates/appsignal.js.erb (hunk 1 above)
Comment (unflxw) [reply support/templates/appsignal.js.erb:13]: Created an issue for the second part: https://github.com/appsignal/appsignal-javascript/issues/598

**Pull request merged**